src/obsolete-reporter.js

```diff
@@ -0,0 +1,64 @@
+const fs = require("fs");
+const path = require("path");
+
+const TOUCHED_FILE_LIST_PATH = path.join(
+  process.cwd(),
+  ".jest-image-snapshot-touched-files"
+);
+
+const IS_ENABLED = !!process.env.JEST_IMAGE_SNAPSHOT_TRACK_OBSOLETE;
+
+class ObsoleteReporter {
+  static isEnabled() {
+    return IS_ENABLED;
+  }
+
+  static markTouchedFile(filePath) {
+    if (!IS_ENABLED) return;
+    const fd = fs.openSync(TOUCHED_FILE_LIST_PATH, "as");
+    fs.writeSync(fd, `${filePath}\n`);
+    fs.closeSync(fd);
+  }
+
+  onRunStart() {
+    if (!IS_ENABLED) return;
+    if (fs.existsSync(TOUCHED_FILE_LIST_PATH)) {
+      fs.unlinkSync(TOUCHED_FILE_LIST_PATH);
+    }
+  }
+
+  onRunComplete(contexts, results) {
+    if (!IS_ENABLED) return;
+    const touchedFiles = Array.from(
+      new Set(
+        fs
+          .readFileSync(TOUCHED_FILE_LIST_PATH, "utf-8")
+          .split("\n")
+          .filter((file) => file && fs.existsSync(file))
+      )
+    );
+    const imageSnapshotDirectories = Array.from(
+      new Set(touchedFiles.map((file) => path.dirname(file)))
+    );
+    const allFiles = imageSnapshotDirectories
+      .map((dir) => fs.readdirSync(dir).map((file) => path.join(dir, file)))
+      .reduce((a, b) => a.concat(b))
+      .filter((file) => file.endsWith("-snap.png"));
+    const obsoleteFiles = allFiles.filter(
+      (file) => !touchedFiles.includes(file)
+    );
+
+    if (fs.existsSync(TOUCHED_FILE_LIST_PATH)) {
+      fs.unlinkSync(TOUCHED_FILE_LIST_PATH);
+    }
+
+    console.log({
```

Comment:
could automatically remove if set to update,

Reply:
the original point was to just print the outdated files for the user to remove them manually, so yes a log would be necessary in that case.

for now, I've updated to automatically remove the files, but that feels slightly riskier IMO.